client/web/compose/src/views/Public/Pages/Records/View.vue

```diff
@@ -383,7 +384,13 @@ export default {
        * came from (and "where" is back).
       */
       if (this.showRecordModal) {
-        this.$bvModal.hide('record-modal')
+        const { recordID, recordPageID } = this.modalPreviousPages.slice(-1)[0] || {}
+
+        if (recordID && recordPageID) {
+          this.$emit('on-modal-back', { recordID, recordPageID, pushModalPreviousPage: false })
```

Comment:
I would always emit this, if its undefined that means it should be closed (and handled in the module)